regulator: cp9314: Adds support for PGOOD pin

rriveramcrus opened the pull request:
Adds support for the PGOOD pin. This pin is asserted by the regulator when converter startup has completed and the output is stable.

rriveramcrus committed:
regulator: cp9314: Adds support for PGOOD pin

Adds support for the PGOOD pin. This pin is asserted by the
regulator when converter startup has completed and the output
is stable. If the PGOOD pin is not used, the PGOOD state is
check via the PGOOD_PIN_STS bit.

Signed-off-by: Ricardo Rivera-Matos <rriveram@opensource.cirrus.com>

kartben commented on drivers/regulator/regulator_cp9314.c:
```diff
@@ -277,6 +279,23 @@ static int regulator_cp9314_enable(const struct device *dev)
 		return ret;
 	}
 
+#ifdef CONFIG_MULTITHREADING
+	k_usleep(CP9314_T_STARTUP_MSEC);
+#else
+	k_busy_wait(CP9314_T_STARTUP_MSEC);
+#endif
```
unless I'm missing something, `k_usleep` already takes care of figuring out whether multi-threading is enable or not?

kartben replied:
I am just surprised that the #ifdef linked in the comment you're mentioning seems to be literally the only place in-tree where such k_sleep vs. k_busy_wait ifdef'ery is happening. But there might very well be a reason that I am completely missing and Andy will clarify :)

aasinclair replied:
Regulators are required to work when `CONFIG_MULTITHREADING=n` so this was necessary to avoid build failure.
However I was not aware that it was fixed here:
https://github.com/zephyrproject-rtos/zephyr/pull/63994

So it is no longer necessary :-)

rriveramcrus replied:
Thanks @aasinclair and @kartben! I'll make sure to clean up the rest of `regulator_cp9314.c` in subsequent PR soon.

rriveramcrus committed:
regulator: cp9314: Adds regulator_get_error_flags support

Adds a handler to check error flag bits. This should be called
by applications that call regulator_enable and receive -EINVAL
as a return.

Signed-off-by: Ricardo Rivera-Matos <rriveram@opensource.cirrus.com>

rriveramcrus committed:
regulator: cp9314: Tidies regulator enable conditional

Enables the converter via I2C as the else case when the EN pin
conditional fails. This used to not matter as no code followed,
but this will change in subsequent patches.

Signed-off-by: Ricardo Rivera-Matos <rriveram@opensource.cirrus.com>

aasinclair commented on drivers/regulator/regulator_cp9314.c:
```diff
@@ -230,6 +236,32 @@ static struct cp9314_reg_patch otp_1_patch[3] = {
 	{CP9314_REG_TSBAT_CTRL, CP9314_LB1_STOP_PHASE_SEL, CP9314_LB1_STOP_PHASE_SEL},
 };
 
+static int regulator_cp9314_check_faults(const struct device *dev)
```
Is the user able to access this information?
Could you link this to the `get_error_flags` API instead?

rriveramcrus commented:
@gmarull gentle reminder on this PR.
